include/dmtcp.h

```diff
@@ -398,6 +398,7 @@ const char *dmtcp_get_ckpt_filename(void) __attribute__((weak));
 const char *dmtcp_get_ckpt_files_subdir(void);
 int dmtcp_should_ckpt_open_files(void);
 int dmtcp_allow_overwrite_with_ckpted_files(void);
+int dmtcp_should_truncate_file_on_restart(const char* path);
```

Comment:
Do we need __attribute__((weak)) just like dmtcp_skip_memory_region_ckpting? 
Ref: https://github.com/dmtcp/dmtcp/blob/9aea31b3dd974f5d2634b3161dfdd9844038bb63/include/procmapsarea.h#L121

Reply:
We don't need weak because we have the terminating function explicitly defined in dmtcpplugin.cpp. We should eventually update other similar APIs as well.